### PR TITLE
115 Fix seeds

### LINKED ISSUE #115: Fix seeds
Running `rake db:seed` goes 💣

### PR DESCRIPTION
**Pull request by mikevallano**
Closes #115 

~Just a couple tweaks to get seeds to run:~ 😆 

- Added `DS_Store` to the `gitignore`. I've since added it to my global `gitignore`, but it doesn't hurt to add it here to make sure any other collaborator doesn't accidentally commit it.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -131,7 +131,7 @@
   {recipe_id: recipe_50.id, quantity: 3.0, measurement_unit: "sprig", name: "fresh cilantro", preparation_style: "chopped" },
   {recipe_id: recipe_50.id, quantity: 1.0, measurement_unit: "T", name: "adobo seasoning", preparation_style: '' },
   {recipe_id: recipe_50.id, quantity: 2.0, measurement_unit: "ounce", name: "queso blanco", preparation_style: "crumbled" },
-  {recipe_id: recipe_50.id, quantity: 2.0, measurement_unit: "can", name: "canned black beans", preparation_style: "finely chopped" },
+  {recipe_id: recipe_50.id, quantity: 20, measurement_unit: "ounce", name: "canned black beans", preparation_style: "finely chopped" },
```
Comment (lortza) [inline db/seeds.rb]: The new `measurement_unit` value you are looking for here is "15oz can" and the `quantity` can stay at "2.0". That should be true for any of the 14 oz can seeds.

File: db/seeds.rb [hunk 2]
```diff
@@ -145,7 +145,7 @@
   {recipe_id: recipe_51.id, quantity: 8.0, measurement_unit: "ounce", name: "spaghetti", preparation_style: "cooked" },
   {recipe_id: recipe_51.id, quantity: 1.0, measurement_unit: "tsp", name: "crushed red pepper", preparation_style: '' },
   {recipe_id: recipe_51.id, quantity: 4.0, measurement_unit: "sprig", name: "fresh cilantro", preparation_style: "chopped" },
-  {recipe_id: recipe_52.id, quantity: 1.0, measurement_unit: "can", name: "coconut milk", preparation_style: '' },
```
Comment (lortza) [inline db/seeds.rb]: Just change the measurement unit to "15oz can" and leave the qty as-is.

File: db/seeds.rb [hunk 3]
```diff
@@ -209,7 +209,7 @@
   {recipe_id: recipe_59.id, quantity: 0.5, measurement_unit: "tsp", name: "granulated garlic", preparation_style: '' },
   {recipe_id: recipe_59.id, quantity: 0.5, measurement_unit: "tsp", name: "sea salt", preparation_style: '' },
   {recipe_id: recipe_59.id, quantity: 1.0, measurement_unit: "T", name: "honey", preparation_style: '' },
-  {recipe_id: recipe_61.id, quantity: 2.0, measurement_unit: "can", name: "black beans", preparation_style: "rinsed and drained" },
+  {recipe_id: recipe_61.id, quantity: 28, measurement_unit: "ounce", name: "black beans", preparation_style: "rinsed and drained" },
```
Comment (lortza) [inline db/seeds.rb]: This one becomes:
```
{ recipe_id: recipe_61.id, quantity: 2, measurement_unit: "15oz can", name: "canned black beans", preparation_style: "rinsed and drained" }
```

File: db/seeds.rb [hunk 4]
```diff
@@ -77,7 +77,7 @@
...
 )
 recipe_85 = Recipe.create!(
-  { user_id: user.id, title: "Sweet Potato Pie", source_name: "ThroughmyiMedia", source_url: "https://youtu.be/BOYWrXVGV5E?t=148", servings: 8, instructions: "Heat oven to 350F\r\n\r\nWash sweet potatoes and poke holes in them with a fork\r\n\r\nRub them in butter and wrap them in foil.\r\n\r\nBake for an hour or until they are soft. Cool them for 20 minutes before peeling. \r\n\r\nPre-bake empty pie shell for 15-20 minutes until dough starts to brown. Remember to cover the edges with foil so they don't brown too fast. \r\n\r\nMash them in a bowl with all of the dry ingredients.\r\n\r\nBeat eggs and add to the potato mix. Incorporate completely. \r\n\r\nAdd butter and milk and mix on slow speed until well incorporated.\r\n\r\nPour mixture into pie shell and bake at 350F for 45 minutes until the pie has set and the crust edges are golden brown.\r\n", prep_time: 0, cook_time: 0, image_url: '', reheat_time: 0, pepperplate_url: '', notes: '', archived: false }
```
Comment (lortza) [inline db/seeds.rb]: ooh yeah good catch. that will bork some stuff.

File: .gitignore [hunk 5]
```diff
@@ -24,3 +24,6 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
```
Comment (lortza) [inline .gitignore]: oh good call!

**Comment by lortza** [comment]
Very nice! In general, the way i'd like the seeds to go now is to use the updated measurement units from here: https://github.com/lortza/food_planner/blob/master/app/models/ingredient.rb#L7

Most of what will need to change in these broken seeds is changing the `measurement_unit` "can" to "15oz can". In some cases, you may need a "30oz can". Don't sweat the difference between 15oz and 14oz cans. 

And honestly, we really don't need that many recipe seeds. I originally wrote these seeds to help populate my production account with actual recipes -- and that's done now. I haven't put any thought into how many recipe seeds we do need, but if it's worth the effort, maybe reduce the number of recipes down to 5. If that's just a bunch of labor, then don't bother. 

Also, when these seeds are all cleaned up and ready to go, please uncomment the line that runs seeds in the CI config: https://github.com/lortza/food_planner/blob/master/.circleci/config.yml#L48 This will force us to be better stewards of this file.

**Comment by mikevallano** [comment]
Changes:

- Move the hard-coded values for recipes & ingredients to YAML files
- Reference the name of the recipe rather than an id
- General cleanup of the seeds file
- Prevent duplicate seeding. If there is already data, the seeds won't run, but will instruct on what to do (can run `rake db:seed:replant` cuz Rails 6 ftw! 🎉 
- Tweak readme for more clarity
- Allow seeds to be run in CI
- Add `DS_Store` to `gitignore`

Screenshot of everything working (first the db is dropped to show what happens from scratch):

seeding

@lortza this is now ready for review

**Comment by lortza** [comment]
@mikevallano There may be a problem with the ingredients yml. If we delete a bunch of recipes, this may break the ingredients.

**Comment by mikevallano** [comment]
@lortza okay it's ready for another look 😅 

Recap of changes:

- Move the hard-coded values for recipes & ingredients to YAML files
- Reference the name of the recipe rather than an id
- General cleanup of the seeds file
- Prevent duplicate seeding. If there is already data, the seeds won't run, but will instruct on what to do (can run `rake db:seed:replant` cuz Rails 6 ftw! 🎉 
- Add a seeds helper module to make the seeds file itself cleaner
- Tweak readme for more clarity
- Allow seeds to be run in CI
- Add `DS_Store` to `gitignore`

File: spec/db/seeds_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
```
Comment (lortza) [inline spec/db/seeds_helper_spec.rb]: Whoa! specs for the seed helper and everything!

File: db/seeds_helper.rb [hunk 7]
```diff
@@ -0,0 +1,52 @@
+module SeedsHelper
+
+  MISSING_RECIPE_WARNING = %(
+    WARNING Ingredients found for recipes that are not in the database.
+    Ingredients for those recipes not created. You should update /db/seed_fixtures/ingredients.yml
+  )
```
Comment (lortza) [inline db/seeds_helper.rb]: very nice! i appreciate getting useful feedback after something fails.

**Pull request merged**